context-free.md

```diff
@@ -2,21 +2,33 @@
 
 ## What is hard to do in this language? Why is it hard?
 
+It's hard to specify complex shapes that aren't recursively defined. This is hard because you'd have to actually specify the shape line by line - consider, for example, the amount of code needed to create the "Welcome" image. It's also hard to visualize what your programs are going to look like without actually compiling them, because it's hard to visualize what the recursion will look like. 
 
 ## How did you learn how to program in this language?
 
+I went through and read the tutorials, then modified the code in the tutorials to see what effects that had on the rendered image. Whenever I wanted to do something but wasn't sure how, I looked it up in the CDFG HOWTO wiki. 
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+As stated above, the programming language is evaluated like a context-free grammar. The parser evaluates the startshape by either drawing the shape or, if it references another rule, substituting in the rule it references. It then evaluates that rule, substituting recursively as relevant. If a rule has multiple subrules, it picks one to use at random. 
+
+One thing I'm not sure about is how it decides to stop making substitutions and just attempt to draw the shape specified by the program - you could certainly imagine a Context Free program where a rule referencing itself (like a fractal) takes forever to evaluate. Presumably the compiler has some way to know when the shape it draws will be smaller than a pixel and then stops evaluating the program then, which is pretty neat. 
+
 
 ## What do you think is interesting about the ContextFree program you wrote?
 
+The ContextFree program I wrote is a modification of the first fractal-generating code ([here](https://web.archive.org/web/20100620122031/http://contextfreeart.org/mediawiki/index.php/Tutorials/Fractals)). I was interested to see what would happen if I changed the base shape for the Serpinski triangle from a TRIANGLE to CIRCLE or SQUARE. I expected this to vastly change what the final image looked like. To my great surprise, the resulting rendered image *still* looked like a Serpinski triangle, just slightly different - for instance, rounder when I made the base image CIRCLE, which is the image I'm submitting with this assignment. 
+
+I also played around with different colours and saturations for a bit, which you can also see in the image I'm submitting. I think changing the saturations for the 3 subtriangles created an interesting visual effect, though it's very subtle when the brightness is turned down.
```

Comment:
I can’t really see it on the triangles themselves with my monitor, but can on the background circle. It is a really neat effect.